core/lib/mobility/plugins/store_based_fallbacks.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'concurrent'
+require 'mobility'
+
+module Mobility
+  class << self
+    attr_reader :store_based_fallbacks
+  end
+
+  @store_based_fallbacks = Concurrent::ThreadLocalVar.new(I18n::Locale::Fallbacks.new)
+
+  module Plugins
+    module StoreBasedFallbacks
+      extend ::Mobility::Plugin
+
+      default true
+      requires :backend, include: :before
+
+      # Applies fallbacks plugin to attributes. Completely disables fallbacks
+      # on model if option is +false+.
+      included_hook do |_, backend_class|
+        unless options[:fallbacks] == false
+          backend_class.include(BackendInstanceMethods)
+        end
+      end
+
+      private
+
+      module BackendInstanceMethods
+        def read(locale, fallback: true, **kwargs)
+          return super(locale, **kwargs) if !fallback || kwargs[:locale]
+
+          locales = fallback == true ? Mobility.store_based_fallbacks.value[locale] : [locale, *fallback]
+          locales.each do |fallback_locale|
+            value = super(fallback_locale, **kwargs)
+            return value if Util.present?(value)
+          end
+
+          super(locale, **kwargs)
```

Comment:
I don't get this part. The method gets a `locale` positional argument, but also a `locale` keyword argument. Shouldn't the logic be something in the lines of:
1. Check whether a translation exists for `locale`
2. If it exists - return it.
3. If it doesn't exist then try to use fallback.

And here I see that the fallback is calculated first, and only then the original locale in the `super(locale, **kwargs)` line. The first line that short-circuits if the kwarg is set is also confusing. What's the difference between the keyword argument and the positional argument? I guess it's some Mobility internal so in our codebase it looks a bit muddy.

Reply:
@kshalot that's what this implementation does. 

The logic is as follows:
1. If a specific locale is requested or fallback is disabled - skip the fallbacks mechanism and read the value without the fallbacks in specified locale
2. Iterate through the fallbacks for requested locale. This list always starts with the locale itself, so e.g. fallbacks['en'] = ['en', (other locales that are configured)]. If a value is found for any of them, return it 
3. If this didn't return any result, read the value without the fallbacks for specified locale

The signature of this function is not the most clear one, but I'd prefer to keep it, not to break other mobility plugins that need to work alongside it

Reply:
I did some further investigation on that, it seems that Mobility's internals call it like `read(locale, locale: true)`. Example here: https://github.com/shioyama/mobility/blob/master/lib/mobility/plugins/active_model/dirty.rb 
So the kwargs[:locale] can be true/false and it has a different semantic. 

This part of the implementation is a slightly modified implementation of the default fallbacks plugin, so I kept the behavior in this case the same.